controllers/multiclusterobservability/observatorium.go

```diff
@@ -33,8 +34,48 @@ const (
 
 	readOnlyRoleName  = "read-only-metrics"
 	writeOnlyRoleName = "write-only-metrics"
+
+	ObservatoriumAPI     = "observatorium-api"
+	ThanosQuery          = "thanos-query"
+	ThanosQueryFrontend  = "thanos-query-frontend"
+	ThanosRule           = "thanos-rule"
+	ThanosReceive        = "thanos-receive-default"
+	ThanosStoreMemcached = "thanos-store-memcached"
+)
+
+var (
+	replicas1      int32 = 1
+	replicas2      int32 = 2
+	replicas3      int32 = 3
+	thanosReplicas       = map[string]*int32{
+		ObservatoriumAPI:    &replicas2,
+		ThanosQuery:         &replicas2,
+		ThanosQueryFrontend: &replicas2,
+
+		ThanosRule:           &replicas3,
+		ThanosReceive:        &replicas3,
+		ThanosStoreMemcached: &replicas3,
+	}
 )
 
+func GetObservatoriumComponentReplicas(componentName string) *int32 {
+	return thanosReplicas[componentName]
+}
+
+func SetObservatoriumComponentReplicas(componentName string, replicas *int32) {
+	for k := range thanosReplicas {
+		// observability-thanos-query-frontend contains thanos-query-frontend and thanos-query
+		if strings.Contains(componentName, ThanosQueryFrontend) {
```

Comment:
why we use this  func `strings.Contains` to check the `componentName`, can not use this like `if k == componentName` to check it.

Reply:
`observability-thanos-query-frontend` contains `thanos-query-frontend` and `thanos-query` so check `thanos-query-frontend` firstly. if contained, then return to avoid make changes on `thanos-query`